detox/src/devices/DeviceRegistry.js

```diff
@@ -0,0 +1,75 @@
+const fs = require('fs');
+const plockfile = require('proper-lockfile');
+const _ = require('lodash');
+const retry = require('../utils/retry');
+const LOCK_FILE = './device.registry.state.lock';
```

Comment:
We are probably not going to support ios+ android sharding/parallelization soon, but we do need to not block users from running them simultaneously. let's create LOCK_FILE per platform.